tests/test_http.py

```diff
@@ -266,3 +269,29 @@ def test_GET_request_chunked_no_content_length(httpserver, chunked_flag):
     assert resp.status_code == 200
     assert "Transfer-encoding" in resp.headers
     assert "Content-length" not in resp.headers
+
+
+def test_httpserver_init_failure_no_stderr_during_cleanup(tmp_path):
+    """
+    Test that, when the server encounters an error during __init__, its cleanup
+    does not raise an AttributeError in its __del__ method, which would emit a 
+    warning onto stderr.
+    """
+
+    script_path = tmp_path.joinpath("script.py")
+
+    script_path.write_text(textwrap.dedent("""
+        from pytest_localserver import http
+        from unittest.mock import patch
+
+        with patch("pytest_localserver.http.make_server", side_effect=RuntimeError("init failure")):
+            server = http.ContentServer()
+    """))
+
+    result = subprocess.run([sys.executable, script_path], stderr=subprocess.PIPE)
```

Comment:
I think you're right. Would you mind kicking off the workflow again?

Reply:
No problem. It's annoying that it requires approval for the workflow every time but I couldn't find a way to prevent that without opening up full permissions for anyone at all to run workflows, and I'm not sure if I should just go ahead and do that (especially considering that this repo is hosted under someone else's organization).